packages/react-utilities/src/hooks/useEffectAfterMount.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useRef, DependencyList, EffectCallback } from 'react';
+
+/**
+ * Similar to React.useEffect, only difference is that the callback is not invoked on mount.
+ *
+ * A common use case is the invocation of external callbacks to ensure controlled component.
+ * @example
+ *  useEffectAfterMount(() => {
+ *    props.onChange?.(value)
+ *  }, [value, props.onChange])
+ * @param callback effect to be invoked when dependencies change
+ * @param dependencies dependency list to determine when the effect should run
+ */
+export function useEffectAfterMount(callback: EffectCallback, dependencies: DependencyList = []) {
```

Comment:
I think adding accordion usage in this PR would also help add more context to the problem

Reply:
This won't probably be used directly by the accordion. The idea of this PR is to implement something similar to [react-ui descendants](https://github.com/reach/reach-ui/tree/develop/packages/descendants), and this hook would be used behind the scenes when callabcks need to be invoked on dependency change but not on mount